r-package/grf/R/input_utilities.R

```diff
@@ -235,17 +235,17 @@ create_data_matrices <- function(X, ..., sample.weights = NULL) {
 }
 
 observation_weights <- function(forest) {
-    sample.weights <- if (is.null(forest$sample.weights)) {
-        rep(1, length(forest$Y.orig))
-    } else {
-        forest$sample.weights * length(forest$Y.orig) / sum(forest$sample.weights)
-    }
-    if (length(forest$clusters) == 0) {
-        observation.weight <- sample.weights
-    } else {
-        clust.factor <- factor(forest$clusters)
-        inverse.counts <- 1 / as.numeric(Matrix::colSums(Matrix::sparse.model.matrix(~ clust.factor + 0)))
-        observation.weight <- sample.weights * inverse.counts[as.numeric(clust.factor)]
-    }
-    observation.weight
+  if (is.null(forest$sample.weights)) {
```

Comment:
I personally like the original way a little better, it follows the classic 'ternary operator' assignment pattern (although maybe it could be formatted more clearly?)